src/core/ui/mainui.cpp

```diff
@@ -14,36 +16,76 @@ namespace otto::ui {
 
     } empty_screen;
 
+    std::map<std::string, std::function<engines::AnyEngine *()>> engines = {
```

Comment:
Sorry, i thought it was possible to just serialize `cur_screen`, but that obviously has no avaliable name. Gotta rethink this then, cause i dont like having  the map here

Reply:
Each of the engines has a "name" method.  However, we'd have to change how the engines are stored/accessed via globals in order to improve this.  Rather than exposing each engine as a field, we'd expose a getEngine() method, which takes an engine name.

Actually, what would be needed is a AnyEngineRegistry. :)